app/controllers/users.js

```diff
@@ -172,6 +176,18 @@ exports.user = (req, res, next, id) => {
       next();
     });
 };
+exports.search = (req, res, next) => {
+  if (isNaN(req.query.q) && req.query.q.trim()) {
```

Comment:
Bug: if q is undefined this will break. `can not read trim of undefined/ undefined does not have a function trim`